signup.js

```diff
@@ -0,0 +1,96 @@
+const emailInput = document.querySelector('#email');
+const passwordInput = document.querySelector('#password');
+const passCheckInput = document.querySelector('#passwordcheck');
+const passEyeButton = document.querySelector('.passEye');
+const checkEyeButton = document.querySelector('.passCheckEye');
+const signButton = document.querySelector('.signbutton');
+
+let A = 0; /* 비밀번호 보이기/숨기기 함수용 변수, 좋은 변수명 구함11 */
+let B = 0; /* 비밀번호 확인 보이기/숨기기 함수용 변수, 좋은 변수명 구함22 */
+
+function loginEmailErrorReset() {
+  document.querySelector('.giveMeEmail').classList.add('hidden');
+  document.querySelector('.notEmail').classList.add('hidden');
+  document.querySelector('.checkEmail').classList.add('hidden');
+  document.querySelector('#email').classList.remove('inputProblem');
+} /* 로그인 - 이메일 오류 메시지를 초기화 하는 함수  */
+
+function loginPassErrorReset() {
+  document.querySelector('.checkPassword').classList.add('hidden');
+  document.querySelector('.giveMePassword').classList.add('hidden');
+  document.querySelector('#password').classList.remove('inputProblem');
+} /* 로그인 - 비밀번호 오류 메시지를 초기화 하는 함수 */
+
+function isEmail(email) {
+  const A = email.indexOf('@');
+  if (A === -1) {
+    return false;
+  } else if (email.slice(0, A).length === 0) {
+    return false;
+  } else if (email.slice(A + 1).length === 0) {
+    return false;
+  }
+  return true;
+} /* 이메일이 맞으면 true를 뱉는 함수*/
+
+const emailPls = function (e) {
+  loginEmailErrorReset();
+  if (emailInput.value === '') {
+    document.querySelector('.giveMeEmail').classList.remove('hidden');
+    document.querySelector('#email').classList.add('inputProblem');
+  } else if (isEmail(emailInput.value) === false) {
+    document.querySelector('.notEmail').classList.remove('hidden');
+    document.querySelector('#email').classList.add('inputProblem');
+  }
+}; /* 이메일을 입력하지 않거나 이메일 주소로 입력하지 않을 때 반응하는 함수 */
+
+const passwordPls = function (e) {
+  loginPassErrorReset();
+  if (passwordInput.value === '') {
+    document.querySelector('.giveMePassword').classList.remove('hidden');
+    document.querySelector('#password').classList.add('inputProblem');
+  }
+}; /* 비밀번호 입력하지 않으면 반응하는 함수 */
+
+emailInput.addEventListener('focusout', emailPls);
+passwordInput.addEventListener('focusout', passwordPls);
+
+passEyeButton.onclick = function () {
+  passEyeButton.firstElementChild.classList.toggle('hidden');
+  passEyeButton.lastElementChild.classList.toggle('hidden');
+  A = A + 1;
+  if (A % 2) {
+    passwordInput.type = 'text';
+  } else {
+    passwordInput.type = 'password';
+  }
+}; /* 비밀번호 칸에서 눈을 눌러 비밀번호 보이기와 숨기기를 하게 하는 함수이자 이벤트 */
+
+signButton.onclick = function (e) {
+  e.preventDefault();
+  loginEmailErrorReset();
+  loginPassErrorReset();
+  if (
+    emailInput.value === 'test@codeit.com' &&
+    passwordInput.value === 'codeit101'
```

Comment:
`test@codeit.com`, `codeit101` 와 같이 항상 값이 고정되어있는 것을 '상수' 라고 합니다
상수가 코드 안에 섞이게 되면 이후 상수를 변경해야 할 때 찾기 힘들기도 하고 관리가 어려워서 
보통 상수는 별도로 분리합니다. 
상수는 대문자 + snake_case 로 작성합니다 -> `SNAKE_CASE` <- 이렇게

예를 들어, 
```
const VALID_EMAIL = 'test@codeit.com';
const VALID_PASSWORD = 'codeit101';
```
와 같이 파일 상단 또는 별도 파일에 작성해두고 여기서는 
```suggestion
    emailInput.value === VALID_EMAIL &&
    passwordInput.value === VALID_PASSWORD
```

로 작성할 수 있습니다!

(참고로 여기서 valid 라는 단어는 '유효한', '통과 가능한' 의 의미로 사용했습니다)